cnf-tests/testsuites/e2esuite/ovs_qos/ovs_qos.go

```diff
@@ -0,0 +1,871 @@
+  labels:
+    machineconfiguration.openshift.io/role: %s
+  name: %s
+spec:
+  config:
+    ignition:
+      version: 2.2.0
+    systemd:
+      units:
+        - contents: |
+            [Unit]
+            Description=Configure ingress bandwidth limiting on br-ex
+            Requires=ovs-configuration.service
+            After=ovs-configuration.service
+            Before=kubelet.service crio.service
+            [Service]
+            Type=oneshot
+            RemainAfterExit=yes
+            ExecStart=/bin/bash -c 'phs=$(/bin/nmcli --get-values GENERAL.DEVICES conn show ovs-if-phys0); /bin/ovs-vsctl set interface $phs ingress_policing_rate=%d; /bin/ovs-vsctl set interface $phs ingress_policing_burst=%d'
+            ExecStop=/bin/bash -c 'phs=$(/bin/nmcli --get-values GENERAL.DEVICES conn show ovs-if-phys0); /bin/ovs-vsctl set interface $phs ingress_policing_rate=0; /bin/ovs-vsctl set interface $phs ingress_policing_burst=0'
+            [Install]
+            WantedBy=multi-user.target
+          enabled: true
+          name: ingress-limit.service
+`, roleWorkerCNF, IngressMCName, ingressRate, ingressRate/10)
+
+	mc, err := createMCWithContent(mcContent, isIngressLimiting)
+	return mc, err
+}
+
+func decodeMCYaml(mcyaml string) (*mcfgv1.MachineConfig, error) {
+	decode := mcfgScheme.Codecs.UniversalDeserializer().Decode
+	obj, _, err := decode([]byte(mcyaml), nil, nil)
+	if err != nil {
+		return nil, err
+	}
+	mc, ok := obj.(*mcfgv1.MachineConfig)
+	if !ok {
+		return nil, fmt.Errorf("couldnt create MC object from mcyaml")
+	}
+
+	return mc, err
+}
+
+func findInitialRate(receiverNode, senderNode string, grabSender bool) (float64, error) {
+	var o iperf3Output
+	var s float64
+	var err error
+
+	senderHostNetwork, receiverHostNetwork, err := networksByConnectivity(sdnToSDN)
+	if err != nil {
+		return 0, err
+	}
+
+	receiverPod := createReceiverPod(receiverNode, receiverHostNetwork)
+	receiverIP := receiverPod.Status.PodIP
+
+	By("Creating sender pod")
+	senderPod := createSenderPod(senderNode, senderHostNetwork, receiverIP)
+
+	if grabSender {
+		o, err = parseIperfOutput(senderPod)
+	} else {
+		o, err = parseIperfOutput(receiverPod)
+	}
+
+	if err != nil {
+		return 0, err
+	}
+
+	for _, sum := range o.Intervals {
+		s += sum.Sum.BitsPerSecond
+	}
+
+	s = s / float64(len(o.Intervals))
+
+	if s == 0 {
+		return 0, fmt.Errorf("rate sum of iperf3 intervals is 0, can't determine rate")
+	}
+
+	return s, nil
+}
+
+func qosOnNode(node *corev1.Node, cmdWithNICTemplate string) (string, error) {
+	ovsPod, err := utilNodes.GetOvsPodByNode(client.Client, node)
+	if err != nil {
+		return "", err
+	}
+
+	nic, err := findOvsPhysicalNic(*ovsPod)
+	if err != nil {
+		return "", err
+	}
+
+	buf, err := pods.ExecCommand(client.Client, *ovsPod, []string{"/bin/bash", "-c", fmt.Sprintf(cmdWithNICTemplate, nic)})
+	if err != nil {
+		return "", err
+	}
+
+	return buf.String(), nil
+}
+
+func findOvsPhysicalNic(ovsPod corev1.Pod) (string, error) {
```

Comment:
please add an example output and explain on the function it will be really hard to debug it